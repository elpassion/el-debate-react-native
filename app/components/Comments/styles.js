import { StyleSheet, Platform } from 'react-native'

export default StyleSheet.create({
  navContainer: {
    position: 'absolute',
    top: 0,
    right: 0,
    left: 0,
    zIndex: 2
  },
  comments: {
    top: (Platform.OS === 'ios') ? 70 : 0,
    left: 0,
    right: 0,
    zIndex: 0,
    height: '100%'
  },
  comment: {
    top: 0,
    height: 80,
    marginLeft: 0,
    marginRight: 15,
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },
  userInitials: {
    position: 'absolute',
    marginTop: 33,
    marginLeft: 25,
    fontFamily: 'helvetica-neue-md',
    fontSize: 12,
    zIndex: 2,
    backgroundColor: 'transparent',
    color: 'white'
  },
  circle: {
    zIndex: 1
  },
  commentBox: {
    marginRight: 80,
    marginTop: 5
  },
  commentUser: {
    position: 'relative',
    marginTop: 15,
    left: -30,
    fontFamily: 'helvetica-neue-md'
  },
  commentText: {
    position: 'relative',
    marginTop: 5,
    left: -30,
    marginRight: 30
  },
  timeBox: {
    marginTop: 20,
    right: 0,
    top: 0,
    position: 'absolute'
  },
  time: {
    color: '#B8B8B8'
  },
  commentDash: {
    borderBottomColor: '#f2f2f2',
    borderBottomWidth: 1,
    width: '90%',
    alignSelf: 'center',
    height: 1
  }
})
