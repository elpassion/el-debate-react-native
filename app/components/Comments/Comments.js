import React, { Component } from 'react'
import { View, ScrollView, Text } from 'react-native';

import styles from './styles'
import Navbar from '/app/components/Shared/Navbar'
import Svg, { Circle } from 'react-native-svg';

export default class Comments extends Component {
  render() {
    const singleComment = (color, isCurrentUserComment, comment, commentTime, userName, userInitials) =>
      <View style={{backgroundColor: isCurrentUserComment ? '#f7f7f7' : 'transparent'}}>
        <View style={styles.comment}>
          <Text style={styles.userInitials}>{userInitials}</Text>
          <Svg height="100" width="100" style={styles.circle}>
            <Circle cx="35" cy="40" r="23" fill={color}/>
          </Svg>
          <View style={styles.commentBox}>
            <Text style={styles.commentUser}>{userName}</Text>
            <Text style={styles.commentText}>{comment}</Text>
          </View>
          <View style={styles.timeBox}>
            <Text style={styles.time}>{commentTime}</Text>
          </View>
        </View>
        <View style={styles.commentDash}/>
      </View>
    return (
      <View>
        <View style={styles.navContainer}>
          <Navbar history={this.props.history}/>
        </View>
        <ScrollView style={styles.comments}>
        </ScrollView>
      </View>
    )
  }
}
