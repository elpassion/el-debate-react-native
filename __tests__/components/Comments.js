import 'react-native';
import React from 'react';
import Comments from 'app/components/Comments/Comments'

it('renders correctly', () => {
  const wrapper = shallow(<Comments />)
  expect(wrapper).toMatchSnapshot();
});
