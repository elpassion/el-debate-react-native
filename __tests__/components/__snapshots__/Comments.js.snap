// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
  >
    <Navbar />
  </View>
  <ScrollViewMock
    style={
      Object {
        "height": "100%",
        "left": 0,
        "right": 0,
        "top": 70,
        "zIndex": 0,
      }
    }
  />
</View>
`;
